actioneval/filesearch.go

```diff
@@ -0,0 +1,296 @@
+		case util.FileTypeEntity:
+			filter.fileType = convertFileType(val)
+		case util.FilterTypeEntity:
+			filter.filterType = convertFilterType(val)
+		case util.FilterGroupId:
+			filter.groupId, err = strconv.ParseInt(val, 10, 64)
+		case util.FilterUserId:
+			filter.userId, err = strconv.ParseInt(val, 10, 64)
+		}
+
+		if err != nil {
+			return nil, err
+		}
+	}
+	return filter, err
+}
+
+func parseMapKeyValToString(item yaml.MapItem) (key string, val string) {
+	key = fmt.Sprintf("%v", item.Key)
+	val = fmt.Sprintf("%v", item.Value)
+	return key, val
+}
+
+func convertFileType(fileType string) util.FileFilterType {
+	switch util.YamlEntityValue(fileType) {
+	case util.DirectoryVal:
+		return util.FileFilterDirectory
+	case util.SymblinkVal:
+		return util.FileFilterSymblink
+	case util.FileVal:
+		return util.FileFilterRegularFile
+	default:
+		return util.FileFilterAll
+	}
+
+}
+
+func convertFilterType(filterType string) util.TextFilterType {
+	switch util.YamlEntityValue(filterType) {
+	case util.ExactVal:
+		return util.TextFilterExact
+	case util.HasPrefixVal:
+		return util.TextFilterHasPrefix
+	case util.HasSuffixVal:
+		return util.TextFilterHasSuffix
+	case util.ContainsVal:
+		fallthrough
+	default:
+		return util.TextFilterContains
+	}
+}
+
+func (f *FileSearchFilter) SearchFilterHandler(workspacePath string, count bool) (result SearchFilterResult) {
+
+	rootPath := path.Join(workspacePath, f.searchLocation)
+	clearRootPath := path.Clean(rootPath)
+
+	// ensure that search location does not escape the workspace
+	if !strings.HasPrefix(clearRootPath, workspacePath) {
+		result.Errmsgs += util.HandleError(fmt.Errorf("relative path "+rootPath+" is not supported "), reflect.TypeOf(f).String())
+		result.State = util.FAIL
+		return result
+	}
+
+	walkMethod := func(filePath string, info os.FileInfo, err error) error {
+
+		loc := path.Join("/", filePath)
+		if !strings.HasPrefix(loc, path.Clean(f.searchLocation)) {
+			return nil
+		}
+
+		if !f.satisfyAllFilters(info) {
+			return nil
+		}
+
+		result.Lines++
+		result.Output.WriteString(loc + "\n")
+		return nil
+	}
+
+	var walkErr error
+	if f.tarHeaders == nil {
+		walkErr = filepath.Walk(clearRootPath, walkMethod)
+	} else {
+		for _, header := range f.tarHeaders {
+			walkErr = walkMethod(header.Name, header.FileInfo(), nil)
+		}
+	}
+
+	if walkErr != nil {
+		result.Errmsgs += util.HandleError(fmt.Errorf(walkErr.Error()), reflect.TypeOf(f).String())
+		result.State = util.FAIL
+		return result
+	}
+	if count {
+		result.Output.Reset()
+		result.Output.WriteString(fmt.Sprintf("%d\n", result.Lines))
+	}
+	return result
+}
+
+func (f *FileSearchFilter) satisfyGroupIdAnUserIdFilter(info os.FileInfo) bool {
+
+	var uid, gid uint32
+	//check groups
+	if info.Sys() != nil {
+
+		if sys, ok := info.Sys().(*syscall.Stat_t); ok {
+			uid = uint32(sys.Uid)
+			gid = uint32(sys.Gid)
+		}
+	}
+
+	if f.userId != -1 && uint32(f.userId) != uid {
```

Comment:
And you could define `f.userId` and `f.groupId` as uint32 so you don't need to cast them either. (It's a good idea to avoid casting if you can, to avoid the possibility of losing information.) 

And then, this code could be simplified by not having the local variables `uid` and `gid` and simply comparing directly against `sys.Uid` and `sys.Gid`